frontend/app/components/inplace-edit/directives/edit-pane/edit-pane.directive.js

```diff
@@ -251,14 +165,30 @@ function InplaceEditorEditPaneController($scope, $element, $location, $timeout,
   });
 
   $scope.$on('form.updateRequired', function() {
-    var submit = $q.defer();
-    vm.updateWorkPackageForm(submit);
+    inplaceEditStorage.updateWorkPackageForm();
   });
 
   $scope.$on('workPackageRefreshed', function() {
     vm.discardEditing();
-    EditableFieldsState.isBusy = false;
+  });
+
+  $scope.$on('inplaceEditMultiStorage.save.workPackage', function (event, promise) {
+    promise.then(function() {
+      $location.hash(null);
```

Comment:
If you're not skipping the promise for non-active fields, you will essentially (try to) focus on all of them. Not sure what will happen in that case. There's also the case of forced edit, where this should not happen.